crates/worker/src/partition/state_machine/mod.rs

```diff
@@ -482,6 +471,70 @@ impl<Codec: RawEntryCodec> StateMachine<Codec> {
+            && !has_idempotency_key
+            && previous_invocation_status != InvocationStatus::Free
+        {
+            // Let's fire a warning in case we got a duplicated invocation id,
+            //  but the invocation is not from the ingress with the same request id!
+            let mut request_from_same_ingress = false;
+            if let Some(ingress_sink @ ServiceInvocationResponseSink::Ingress { .. }) =
+                &service_invocation.response_sink
+            {
+                if let InvocationStatus::Scheduled(ScheduledInvocation {
+                    metadata: PreFlightInvocationMetadata { response_sinks, .. },
+                })
+                | InvocationStatus::Inboxed(InboxedInvocation {
+                    metadata: PreFlightInvocationMetadata { response_sinks, .. },
+                    ..
+                })
+                | InvocationStatus::Invoked(InFlightInvocationMetadata {
+                    response_sinks,
+                    ..
+                })
+                | InvocationStatus::Suspended {
+                    metadata: InFlightInvocationMetadata { response_sinks, .. },
+                    ..
+                } = &previous_invocation_status
+                {
+                    if response_sinks.contains(ingress_sink) {
+                        request_from_same_ingress = true;
+                    }
+                }
+            }
+
+            if let Source::Ingress(_) = &service_invocation.source {
+                if let InvocationStatus::Scheduled(ScheduledInvocation {
+                    metadata: PreFlightInvocationMetadata { source, .. },
+                })
+                | InvocationStatus::Inboxed(InboxedInvocation {
+                    metadata: PreFlightInvocationMetadata { source, .. },
+                    ..
+                })
+                | InvocationStatus::Invoked(InFlightInvocationMetadata { source, .. })
+                | InvocationStatus::Suspended {
+                    metadata: InFlightInvocationMetadata { source, .. },
+                    ..
+                }
+                | InvocationStatus::Completed(CompletedInvocation { source, .. }) =
+                    &previous_invocation_status
+                {
+                    if service_invocation.source == *source {
+                        request_from_same_ingress = true;
+                    }
+                }
+            }
+
+            if !request_from_same_ingress {
+                warn!(
+                    "Detected a potential bug here: \
+                    I got a duplicated invocation id, for an invocation that is not idempotent (neither has idempotency key, nor it's workflow run), \
+                    and the source of the request is not the same RPC. \
+                    This seems like either a bug in the internal message routing, or a ULID collision, meaning no one would be reading this message because an asteroid already hit earth.");
+            }
```

Comment:
I'm afraid this will trigger also if the user goes first back and then forward with versions, now that I made the Source::Ingress parsing return a random id, in case one is not present.

Reply:
So i'm actually inclined to just remove it...

Reply:
I think removing it should be fine.